builtin/credential/aws-ec2/path_login.go

```diff
@@ -288,6 +295,15 @@ func (b *backend) pathLoginUpdate(
 		}
 	}
 
+	// If the clientNonce is empty at this point, it means that its a first
+	// time login and that a nonce is not supplied. Create a random nonce
+	// to be associated for the instance ID.
+	if clientNonce == "" {
```

Comment:
Unfortunately not. The final value of DisallowReauthentication is determined only after role-tag is processed.